pktoken/pktoken.go

```diff
@@ -51,6 +63,119 @@ func FromCompact(pktCom []byte) (*PKToken, error) {
 	}
 }
 
+func (p *PKToken) ToCompact() []byte {
+	if p.Payload == nil {
+		panic(fmt.Errorf("Payload can not be nil"))
+	}
+
+	var buf bytes.Buffer
+	buf.WriteString(string(p.Payload))
+	buf.WriteByte(':')
+	buf.WriteString(string(p.OpPH))
+	buf.WriteByte(':')
+	buf.WriteString(string(p.OpSig))
+	buf.WriteByte(':')
+	buf.WriteString(string(p.CicPH))
+	buf.WriteByte(':')
+	buf.WriteString(string(p.CicSig))
+
+	if p.CosPH != nil {
+		buf.WriteByte(':')
+		buf.WriteString(string(p.CosPH))
+		buf.WriteByte(':')
+		buf.WriteString(string(p.CosSig))
+	}
+
+	pktCom := buf.Bytes()
+	return pktCom
+}
+
+func FromJWS(jws *JWS) *PKToken {
+	var cic, op, cos JWSignature
+
+	gq := false
+	for _, sig := range jws.Signatures {
+		switch sig.Public["sig_type"] {
+		case "oidc":
+			op = sig
+		case "oidc_gq":
+			op = sig
+			gq = true
+		case "cic":
+			cic = sig
+		}
+	}
+
+	return &PKToken{
+		Payload: []byte(jws.Payload),
+		OpPH:    []byte(op.Protected),
+		OpSig:   []byte(op.Signature),
+		OpSigGQ: gq,
+		CicPH:   []byte(cic.Protected),
+		CicSig:  []byte(cic.Signature),
+		CosPH:   []byte(cos.Protected),
+		CosSig:  []byte(cos.Signature),
+	}
+}
+
+func (p *PKToken) ToJWS() *JWS {
+	var opSignType string
+	if p.OpSigGQ {
+		opSignType = "oidc_gq"
```

Comment:
Might be worth pulling the signing type strings out and making it an enum/constant so that the string only needs to be defined in one place

Reply:
Good call, done!